### PR TITLE
AP_Landing: enable deepstall landing to be compiled out of the code

### PR DESCRIPTION
**Pull request by peterbarker**
Saves ~5.5kB

This library really needs a proper frontend/backend split.

### CODE REVIEW

**Comment by peterbarker** [comment]
We probably want a prearm check that the landing type configured in `LAND_TYPE` is valid before this goes in.

**Review by khancyr — APPROVED** [review]
LGTM

File: libraries/AP_Landing/AP_Landing_Deepstall.cpp [hunk 1]
```diff
@@ -655,3 +658,5 @@ float AP_Landing_Deepstall::update_steering()
 
     return ds_PID.get_pid(error);
 }
+
+#endif
```
Comment (tridge) [inline libraries/AP_Landing/AP_Landing_Deepstall.cpp:662]: add // HAL_LANDING_DEEPSTALL_ENABLED

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Landing/AP_Landing_Deepstall.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Landing/AP_Landing_Deepstall.cpp]: Done

**Pull request merged**